feature-libs/cart/base/components/added-to-cart-toast/added-to-cart-toast.component.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  HostBinding,
+  OnDestroy,
+  Renderer2,
+} from '@angular/core';
+import { WindowRef } from '@spartacus/core';
+import { Observable, OperatorFunction, Subscription } from 'rxjs';
+import { filter, take } from 'rxjs/operators';
+import {
+  ActiveCartFacade,
+  CartToastItem,
+  CartUiEventAddToCart,
+  CART_TOAST_STATE,
+  OrderEntry,
+} from '@spartacus/cart/base/root';
+import { AddedToCartToastComponentService } from './added-to-cart-toast-component.service';
+
+@Component({
+  selector: 'cx-added-to-cart-toast',
+  templateUrl: './added-to-cart-toast.component.html',
+})
+export class AddedToCartToastComponent implements OnDestroy {
+  protected subscription = new Subscription();
+
+  headerElement: HTMLElement | null;
+
+  toastContainerClass: string;
+
+  timeout: number;
+
+  toastContainerBaseClass = 'toast-container';
+
+  lastToastItem: CartToastItem;
+
+  scrollEventUnlistener: () => void;
+
+  @HostBinding('className') baseClass: string;
+
+  cartToasts$: Observable<CartToastItem[]> =
+    this.addedToCartToastService.getToasts();
+
+  constructor(
+    protected winRef: WindowRef,
+    protected renderer: Renderer2,
+    protected cd: ChangeDetectorRef,
+    protected addedToCartToastService: AddedToCartToastComponentService,
+    protected activeCartService: ActiveCartFacade,
+    protected el: ElementRef
+  ) {}
+
+  init(): void {
+    this.baseClass = 'cx-added-to-cart-toast';
+    this.toastContainerClass = this.toastContainerBaseClass;
+    this.headerElement = this.winRef.document.querySelector('header');
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
+  addToast(event: CartUiEventAddToCart) {
+    this.setOffsetHeight();
+
+    if (this.lastToastItem) {
+      this._destroyPreviousToast();
+      this.lastToastItem = null;
+    }
+
+    this.activeCartService
+      .getLastEntry(event.productCode)
+      .pipe(
+        filter(Boolean) as OperatorFunction<OrderEntry | undefined, OrderEntry>,
+        take(1)
+      )
+      .subscribe((entries) => {
+        let toastContainerClass = `${this.toastContainerBaseClass} `;
+        const toastItem = this.addedToCartToastService.addToast(
+          event.quantity,
+          entries,
+          toastContainerClass
+        );
+
+        this.cd.detectChanges();
+
+        // trigger enter animations with class change
+        toastItem.baseClass = this.getToastStyles();
+
+        if (this.scrollEventUnlistener) {
+          this.scrollEventUnlistener();
+        }
+
+        // wait for enter animation end
+        setTimeout(() => {
```

Comment:
There are 2 timeouts in the component an a third on in the component service.  This seems like a lot of timeouts for a toast that simply needs to disappear after a given delay.  Are they all required?  Can the implementation be simplified?